src/Microsoft.Extensions.Localization/ResourceManagerStringLocalizerFactory.cs

```diff
@@ -64,7 +64,16 @@ public IStringLocalizer Create(Type resourceSource)
 
             var typeInfo = resourceSource.GetTypeInfo();
             var assembly = typeInfo.Assembly;
-            var baseName = _applicationEnvironment.ApplicationName + "." + _resourcesRelativePath + resourceSource.Name;
+
+            string baseName = string.Empty;
+            if (!string.IsNullOrEmpty(_resourcesRelativePath))
```

Comment:
Can you use this
```csharp
baseName = $"{_applicationEnvironment}.{_resourcesRelativePath}{typeInfo.FullName}";
```
instead, because if the ```resourcesRelativePath``` is null or empty the value will be nothing. I used a string interpolation to make things pretty :smile: